reports/views.py

```diff
@@ -1 +1,45 @@
-# Create your views here.
+from django.conf import settings
+from django.utils import timezone
+from rest_framework.exceptions import ValidationError
+from rest_framework.permissions import IsAdminUser
+from rest_framework.response import Response
+from rest_framework.views import APIView
+
+from reports.tasks import generate_report
+from reports.utils import midnight, midnight_validator, one_month_after
+
+
+class ReportsView(APIView):
+
+    """ Reports Generation
+        POST - starts up the task that generates the reports
+    """
+    permission_classes = (IsAdminUser,)
+
+    def post(self, request, *args, **kwargs):
+        try:
+            output_file = request.data['output_file']
+        except KeyError:
+            raise ValidationError("Please specify 'output_file'.")
+        if 'start_date' in request.data:
+            start_date = midnight_validator(request.data['start_date'])
+        else:
+            start_date = midnight(timezone.now())
+        if 'end_date' in request.data:
+            end_date = midnight_validator(request.data['end_date'])
+        else:
+            end_date = one_month_after(start_date)
+        email_to = request.data.get('email_to', [])
+        email_from = request.data.get(
+            'email_from', settings.DEFAULT_FROM_EMAIL)
+        email_subject = request.data.get(
+            'email_subject', 'Seed Control Interface Generated Report')
```

Comment:
What about creating a serializer for this, and having that do the checking for us? http://www.django-rest-framework.org/api-guide/serializers/